### PR TITLE
python3Packages.subarulink: init at 0.3.11

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Python module for interacting with STARLINK-enabled vehicle.

https://github.com/G-Two/subarulink

This is a Home Assistant (HACS) dependency.

This add in addition [`stdiomask`](https://github.com/asweigart/stdiomask).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/subarulink/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+, pytest-asyncio
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "subarulink";
+  version = "0.3.11";
+
+  src = fetchFromGitHub {
+    owner = "G-Two/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/subarulink/default.nix:18]:
```suggestion
    owner = "G-Two";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108724` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.stdiomask</li>
<li>python37Packages.subarulink</li>
<li>python38Packages.stdiomask</li>
<li>python38Packages.subarulink</li>
<li>python39Packages.stdiomask</li>
<li>python39Packages.subarulink</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
darwin patch

```diff
diff --git a/pkgs/development/python-modules/subarulink/default.nix b/pkgs/development/python-modules/subarulink/default.nix
index cb7ac728696..4d88604a51c 100644
--- a/pkgs/development/python-modules/subarulink/default.nix
+++ b/pkgs/development/python-modules/subarulink/default.nix
@@ -31,6 +31,8 @@ buildPythonPackage rec {
     pytestCheckHook
   ];

+   __darwinAllowLocalNetworking = true;
+
   pythonImportsCheck = [ "subarulink" ];

   meta = with lib; {
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108724` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.subarulink: log <a href="https://termbin.com/le5v">https://termbin.com/le5v</a></li>
<li>python38Packages.subarulink: log <a href="https://termbin.com/fri8">https://termbin.com/fri8</a></li>
<li>python39Packages.subarulink: log <a href="https://termbin.com/b17s">https://termbin.com/b17s</a></li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.stdiomask</li>
<li>python38Packages.stdiomask</li>
<li>python39Packages.stdiomask</li>
</ul>
</details>

File: pkgs/development/python-modules/subarulink/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/subarulink/default.nix]: Thanks, updated.

**Comment by fabaff** [comment]
> darwin patch

added. Thanks

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -7219,6 +7221,8 @@ in {
 
   subdownloader = callPackage ../development/python-modules/subdownloader { };
 
+  subarulink = callPackage ../development/python-modules/subarulink { };
```
Comment (mweinelt) [inline pkgs/top-level/python-packages.nix:7224]: suba before subd.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Looks fine, one nit.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108724` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.stdiomask</li>
<li>python37Packages.subarulink</li>
<li>python38Packages.stdiomask</li>
<li>python38Packages.subarulink</li>
<li>python39Packages.stdiomask</li>
<li>python39Packages.subarulink</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108724` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.stdiomask</li>
<li>python37Packages.subarulink</li>
<li>python38Packages.stdiomask</li>
<li>python38Packages.subarulink</li>
<li>python39Packages.stdiomask</li>
<li>python39Packages.subarulink</li>
</ul>
</details>

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/top-level/python-packages.nix:7222]: Thanks, fixed.

**Pull request merged**